### PR TITLE
New rule types support automated response actions

### PR DESCRIPTION
**Pull request by joepeeples**
> [!IMPORTANT]
> - Wait until https://github.com/elastic/kibana/pull/193390 is merged AND promoted in serverless production for feature to become available in the app.

Resolves https://github.com/elastic/security-docs-internal/issues/40. Updates both serverless and 8.16 ESS docs.

## Previews
- Serverless: Click the link in the [comment below](https://github.com/elastic/security-docs/pull/5797#issuecomment-2341132802), then in the Security card click **View serverless docs**. Navigate to: **Endpoint response actions → Automated response actions**
- ESS: [Automated response actions](https://security-docs_bk_5797.docs-preview.app.elstc.co/guide/en/security/master/automated-response-actions.html)

### CODE REVIEW

**Review by benironside — APPROVED** [review]
LGTM!

**Review by natasha-moore-elastic — APPROVED** [review]

File: docs/management/admin/automated-response-actions.asciidoc [hunk 1]
```diff
@@ -14,13 +14,13 @@ Add {elastic-defend}'s <<response-actions,response actions>> to detection rules
 * Automated response actions require an https://www.elastic.co/pricing[Enterprise subscription].
 * Hosts must have {agent} installed with the {elastic-defend} integration.
 * Your user role must have the ability to create detection rules and the <<endpoint-management-req,privilege>> to perform specific response actions (for example, the **Host Isolation** privilege to isolate hosts).
-* You can only add automated response actions to custom query rules.
+* You can only add automated response actions to <<create-custom-rule,custom query>>, <<create-eql-rule,event correlation (EQL)>>, and <<create-esql-rule,{esql}>> type rules.
```
Comment (tomsonpl) [inline docs/management/admin/automated-response-actions.asciidoc:17]: In the end we introduced this also to new-terms rule. Could you update this too? Sorry for late info! Thank you 🙇

Comment (joepeeples) [reply docs/management/admin/automated-response-actions.asciidoc]: No problem, added in https://github.com/elastic/security-docs/pull/5797/commits/99f7c431aa060d2a912e1e8c8d72b6966a4c3e54. Thanks for the heads-up!

File: docs/management/admin/automated-response-actions.asciidoc [hunk 2]
```diff
@@ -14,13 +14,13 @@ Add {elastic-defend}'s <<response-actions,response actions>> to detection rules
 * Automated response actions require an https://www.elastic.co/pricing[Enterprise subscription].
 * Hosts must have {agent} installed with the {elastic-defend} integration.
 * Your user role must have the ability to create detection rules and the <<endpoint-management-req,privilege>> to perform specific response actions (for example, the **Host Isolation** privilege to isolate hosts).
-* You can only add automated response actions to custom query rules.
+* You can only add automated response actions to <<create-custom-rule,custom query>>, <<create-eql-rule,event correlation (EQL)>>, <<create-new-terms-rule,new terms>>,and <<create-esql-rule,{esql}>> type rules.
```
Comment (benironside) [inline docs/management/admin/automated-response-actions.asciidoc:17]:
```suggestion
* You can only add automated response actions to <<create-custom-rule,custom query>>, <<create-eql-rule,event correlation (EQL)>>, <<create-new-terms-rule,new terms>>, and <<create-esql-rule,{esql}>> type rules.
```

**Review by benironside — COMMENTED** [review]
one minor thing! Otherwise LGTM

**Review by benironside — APPROVED** [review]

**Review by caitlinbetz — DISMISSED** [review]

**Review by nastasha-solomon — APPROVED** [review]

**Pull request merged**